ietf/doc/utils.py

```diff
@@ -491,16 +509,19 @@ def update_action_holders(doc, prev_state=None, new_state=None, prev_tags=None,
             # Default to responsible AD for states other than these
             doc.action_holders.add(doc.ad)
     
-    if prev_tag_slugs != new_tag_slugs:
-        # If we have added or removed the need-rev tag, add or remove authors as action holders
-        if ('need-rev' in prev_tag_slugs) and ('need-rev' not in new_tag_slugs):
-            # Removed the 'need-rev' tag - drop authors from the action holders list
-            DocumentActionHolder.objects.filter(document=doc, person__in=doc.authors()).delete()
-        elif ('need-rev' not in prev_tag_slugs) and ('need-rev' in new_tag_slugs):
-            # Added the 'need-rev' tag - add authors to the action holders list
-            for auth in doc.authors():
-                if not doc.action_holders.filter(pk=auth.pk).exists():
-                    doc.action_holders.add(auth)
+    # If we have added or removed the need-rev tag, add or remove authors as action holders
+    if tags.removed("need-rev"):
+        # Removed the 'need-rev' tag - drop authors from the action holders list
+        DocumentActionHolder.objects.filter(document=doc, person__in=doc.authors()).delete()
+    elif tags.added("need-rev"):
```

Comment:
Only a few seconds thought given to it, but I'm not seeing a case where this tag being added shouldn't relieve the AD of being an action holder. (In case that's too many nested nots, I think the ad should be removed as an action holder when this tag is added.

@larseggert - am I missing any use cases that say the AD should remain as an action holder when this tag is added?

Reply:
Sounds reasonable to me - I'll add

Reply:
When _which_ tag is added?

Reply:
the "need-rev" (identified as the "Revised I-D needed" in the UI)

Reply:
When that is added, IMO the AD should never remain an action holder.